File: school-management-frontend/src/services/teacherService.js
import api from '../api/axios';

const teacherService = {
  getAllTeachers: () => api.get('/teachers'),
  getTeacherById: (id) => api.get(`/teachers/${id}`),
  createTeacher: (teacherData) => api.post('/teachers', teacherData),
  updateTeacher: (id, teacherData) => api.put(`/teachers/${id}`, teacherData),
  deleteTeacher: (id) => api.delete(`/teachers/${id}`),
};

export default teacherService;

File: school-management-backend/src/main/java/com/school/management/service/TeacherService.java
package com.school.management.service;

import com.school.management.dto.TeacherRequest;
import com.school.management.entity.Teacher;

import java.util.List;

public interface TeacherService {
    List<Teacher> getAllTeachers();
    Teacher getTeacherById(Long id);
    Teacher createTeacher(TeacherRequest request);
    Teacher updateTeacher(Long id, TeacherRequest request);
    void deleteTeacher(Long id);
}
